src/metabase/channel/email.clj

```diff
@@ -111,17 +114,34 @@
 
 (defn- notification-recipients->emails
   [recipients]
-  (for [recipient recipients]
-    (case (:type recipient)
-      :notification-recipient/user
-      (-> recipient :user :email))))
+  (flatten (for [recipient recipients
```

Comment:
`(into [] cat` maybe? `flatten` feels like smell :)

Reply:
I'm not against `into` but why `flatten` is a smell?

Reply:
Mostly because it flattens everything recursively, so it feels a bit like "IDK what's going on here" in regards to data structures.

Reply:
Yeah, flatten doesn't convey intent and is brittle to the data shape changing. It's also slower 😄

Reply:
Oh, right, I didn't realize that it does that recursively, makes sense, then!